spec/dynamoid/criteria_spec.rb

```diff
@@ -38,14 +38,16 @@
   end
 
   it 'passes each to all members' do
-    User.each do |u|
-      expect(u.id == user1.id || u.id == user2.id).to be_truthy
-      expect(u.new_record).to be_falsey
-    end
+    expect { |b| User.each(&b)}.to yield_successive_args(
+      be_kind_of(User).and(have_attributes('new_record' => false)),
+      be_kind_of(User).and(have_attributes('new_record' => false))
+    )
   end
 
   it 'passes find_by_pages to all members' do
-    expect(User.find_by_pages).to match([contain_exactly(user1, user2)])
+    expect {|b| User.find_by_pages(&b)}.to yield_successive_args(
+      [all(be_kind_of(User)), {last_evaluated_key: nil}],
+    )
```

Comment:
We need to have spec to test that returned `last_evaluated_key` is a valid argument for `start` method and paging actually works. Neither of new specs check something except it's a Hash instance.

Does it make sense?

Reply:
Yep, I'll find somewhere to put that end-to-end(ish).